grails-app/domain/org/pih/warehouse/order/OrderAdjustment.groovy

```diff
@@ -104,6 +104,22 @@ class OrderAdjustment implements Serializable, Comparable<OrderAdjustment> {
         return !postedPurchaseInvoiceItems.empty
     }
 
+    OrderAdjustmentInvoiceStatus getDerivedPaymentStatus() {
+        Boolean fullyInvoiced = invoicedUnitPrice == totalAdjustments
+
+        if (isInvoiced) {
+            if (canceled || totalAdjustments == 0) {
+                return OrderAdjustmentInvoiceStatus.INVOICED
+            }
+            if (fullyInvoiced) {
+                return OrderAdjustmentInvoiceStatus.INVOICED
+            }
+            return OrderAdjustmentInvoiceStatus.PARTIALLY_INVOICED
+        } else {
+            return OrderAdjustmentInvoiceStatus.NOT_INVOICED
+        }
+    }
```

Comment:
Wouldn't it be cleaner to check:
```
if (!isInvoiced) {
return OrderAdjustmentInvoiceStatus.NOT_INVOICED
}
```
at the beginning, and then the the rest of your conditions?